crates/net/discv4/src/lib.rs

```diff
@@ -1943,6 +1941,40 @@ mod tests {
         .await
     }
 
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_random_lookup() {
+        reth_tracing::init_tracing();
+
+        let config = Discv4Config::builder().build();
+        let (_discv4, mut service) = create_discv4_with_config(config).await;
+
+        let target = PeerId::random();
+
+        let id = PeerId::random();
+        let key = kad_key(id);
+        let record = NodeRecord::new("0.0.0.0:0".parse().unwrap(), id);
+
+        let _ = service.kbuckets.insert_or_update(
+            &key,
+            NodeEntry::new(record),
+            NodeStatus {
+                direction: ConnectionDirection::Incoming,
+                state: ConnectionState::Connected,
+            },
+        );
+
+        service.lookup(target);
+        assert_eq!(service.pending_find_nodes.len(), 1);
+
+        let ctx = service.pending_find_nodes.values().next().unwrap().lookup_context.clone();
+
+        assert_eq!(ctx.target(), target);
+        assert_eq!(ctx.inner.closest_nodes.borrow().len(), 1);
+
+        ctx.add_node(record);
+        assert_eq!(ctx.inner.closest_nodes.borrow().len(), 1);
```

Comment:
So previously the call to `lookup` would [first initialize the distance relative to the `target`](https://github.com/paradigmxyz/reth/blob/ade1ea4216136e05b02585e1168ea154fd08988f/crates/net/discv4/src/lib.rs#L617-L625):
```rust
        let key = kad_key(target);

        // Start a lookup context with the 16 (MAX_NODES_PER_BUCKET) closest nodes
        let ctx = LookupContext::new(
            target,
            self.kbuckets
                .closest_values(&key)
                .take(MAX_NODES_PER_BUCKET)
                .map(|n| (key.distance(&n.key), n.value.record)),
```
So we have `DistanceOne -> QueryNode { record, queried: true, .. }`
And then, in `add_node`, [we would have used our own key to calculate the distance](https://github.com/paradigmxyz/reth/blob/ade1ea4216136e05b02585e1168ea154fd08988f/crates/net/discv4/src/lib.rs#L1119-L1121):
```rust
            let key = kad_key(node.id);
            let distance = our_key.distance(&key);
            ctx.add_node(distance, node);
```
Which would add another mapping `DistanceTwo -> QueryNode { record, queried: false, .. }`, meaning the previous behavior would return `2` here. good test!